runtime/parachains/src/hrmp.rs

```diff
@@ -2009,3 +2140,251 @@ mod tests {
+			assert_ok!(Configuration::<T>::set_hrmp_max_parachain_outbound_channels(frame_system::RawOrigin::Root.into(), e + 1));
+			assert_ok!(Configuration::<T>::set_hrmp_max_parachain_inbound_channels(frame_system::RawOrigin::Root.into(), i + 1));
+			// .. and enact it.
+			Configuration::<T>::initializer_on_new_session(&Shared::<T>::scheduled_session());
+
+			let config = Configuration::<T>::config();
+			let deposit: BalanceOf<T> = config.hrmp_sender_deposit.unique_saturated_into();
+
+			let para: ParaId = 1u32.into();
+			let para_origin: crate::Origin = 1u32.into();
+			register_parachain_with_balance::<T>(para, deposit);
+			T::Currency::make_free_balance_be(&para.into_account(), deposit * 256u32.into());
+
+			for ingress_para_id in 0..i {
+				// establish ingress channels to `para`.
+				let ingress_para_id = ingress_para_id + PREFIX_0;
+				let _ = establish_para_connection::<T>(ingress_para_id, 1, ParachainSetupStep::Established);
+			}
+
+			// nothing should be left unprocessed.
+			assert_eq!(HrmpOpenChannelRequestsList::<T>::decode_len().unwrap_or_default(), 0);
+
+			for egress_para_id in 0..e {
+				// establish egress channels to `para`.
+				let egress_para_id = egress_para_id + PREFIX_1;
+				let _ = establish_para_connection::<T>(1, egress_para_id, ParachainSetupStep::Established);
+			}
+
+			// nothing should be left unprocessed.
+			assert_eq!(HrmpOpenChannelRequestsList::<T>::decode_len().unwrap_or_default(), 0);
+
+			// all in all, we have created these many channels
+			assert_eq!(HrmpChannels::<T>::iter().count() as u32, i + e);
+		}: _(frame_system::Origin::<T>::Root, para, i, e) verify {
+			// all in all, all of them must be gone by now.
+			assert_eq!(HrmpChannels::<T>::iter().count() as u32, 0);
+		}
+
+		force_process_hrmp_open {
+			// number of channels that need to be processed. Worse case is an N-M relation: unique
+			// sender and recipients for all channels.
+			let c in 0 .. MAX_UNIQUE_CHANNELS;
+
+			for id in 0 .. c {
+				let _ = establish_para_connection::<T>(PREFIX_0 + id, PREFIX_1 + id, ParachainSetupStep::Accepted);
+			}
+			assert_eq!(HrmpOpenChannelRequestsList::<T>::decode_len().unwrap_or_default() as u32, c);
+		}: _(frame_system::Origin::<T>::Root, c)
+		verify {
+			assert_eq!(HrmpOpenChannelRequestsList::<T>::decode_len().unwrap_or_default() as u32, 0);
+		}
+
+		force_process_hrmp_close {
+			// number of channels that need to be processed. Worse case is an N-M relation: unique
+			// sender and recipients for all channels.
+			let c in 0 .. MAX_UNIQUE_CHANNELS;
+
+			for id in 0 .. c {
+				let _ = establish_para_connection::<T>(PREFIX_0 + id, PREFIX_1 + id, ParachainSetupStep::CloseRequested);
+			}
+
+			assert_eq!(HrmpCloseChannelRequestsList::<T>::decode_len().unwrap_or_default() as u32, c);
+		}: _(frame_system::Origin::<T>::Root, c)
+		 verify {
+			 assert_eq!(HrmpCloseChannelRequestsList::<T>::decode_len().unwrap_or_default() as u32, 0);
+		 }
```

Comment:
nit: Extra space at the beginning:

```suggestion
		verify {
			assert_eq!(HrmpCloseChannelRequestsList::<T>::decode_len().unwrap_or_default() as u32, 0);
		}
```